specifyweb/frontend/js_src/lib/components/Forms/DeleteBlocked.tsx

```diff
@@ -140,6 +142,40 @@ function BlockerPreview({
     [ids]
   );
   const table = parentRelationship?.relatedModel ?? directRelationship.model;
+  const resolvedOthersideQuery = React.useMemo(() => {
+    /*
+     * Check if parent relationship exists. If not, optimize via direct relationship query.
+     * If exists, and othersidename is valid, optimize via other side.
+     * Otherwise, default to in query
+     *
+     */
+    const rawQueryField =
+      parentRelationship === undefined
+        ? QueryFieldSpec.fromPath(table.name, [
+            directRelationship.name,
+            directRelationship.relatedModel.idField.name,
+          ])
+        : parentRelationship.otherSideName !== undefined
+        ? QueryFieldSpec.fromPath(table.name, [
+            parentRelationship.otherSideName,
+            directRelationship.name,
+            directRelationship.relatedModel.idField.name,
+          ])
+        : undefined;
+
+    return (
+      rawQueryField
+        ?.toSpQueryField()
+        .set('isDisplay', false)
+        .set('operStart', queryFieldFilters.equal.id)
+        .set('startValue', parentResource.id.toString()) ??
+      QueryFieldSpec.fromPath(table.name, ['id'])
+        .toSpQueryField()
+        .set('isDisplay', false)
+        .set('startValue', resolvedIds.join(','))
+        .set('operStart', queryFieldFilters.in.id)
+    );
+  }, [resolvedIds]);
```

Comment:
If I am not mistaken, this code is the "bulk" of this Pull Request. 
Looks good!

Reply:
yup, that's where it all is.